crates/shared/src/api.rs

```diff
@@ -123,6 +127,190 @@ impl IntoWarpReply for anyhowError {
     }
 }
 
+impl IntoWarpReply for CalculateQuoteError {
```

Comment:
I think we should keep the `IntoWarpReply` implementations in the `orderbook` crate - its specific about promoting internal errors into HTTP API RPC responses.